frontend/packages/console-dynamic-plugin-sdk/docs/api.md

```diff
@@ -2077,6 +2078,46 @@ Component that allows to receive contributions from other plugins for the `conso
 
 
 
+---
+
+## `NamespaceBar`
+
+### Summary 
+
+A component that renders a horizontal toolbar with a namespace dropdown menu in the leftmost position. Additional components can be passed in as children and will be rendered to the right of the namespace dropdown. This component is designed to be used at the top of the page. It should be used on pages where the user needs to be able to change the active namespace, such as on pages with k8s resources.
+
+
+
+### Example
+
+
+```tsx
+   const logNamespaceChange = (namespace) => console.log(`New namespace: ${namespace}`);
+
+   ...
+
+   <NamespaceBar onNamespaceChange={logNamespaceChange}>
+     <NamespaceBarApplicationSelector />
+   </NamespaceBar>
+   <Page>
+
+     ...
+```
+
+
+
+
+
+### Parameters
+
+| Parameter Name | Description |
+| -------------- | ----------- |
+| `onNamespaceChange` | (optional) A function that is executed when a namespace option is selected. It accepts the new namespace in the form of a string as its only argument. The active namespace is updated automatically when an option is selected, but additional logic can be applied via this function. When the namespace is changed, the namespace parameter in the URL will be changed from the previous namespace to the newly selected namespace. |
+| `isDisabled` | (optional) A boolean flag that disables the namespace dropdown if set to true. This option only applies to the namespace dropdown and has no effect on child components. |
```

Comment:
Is this use case for this that you want to show other dropdowns and hide just the projects dropdown? I think it would be better to remove this from the public API even if it's used internally. At best, the prop is misnamed since it's hiding the dropdown, not disabling it.

@christianvogt or @jerolimov might know better how this is used.

Reply:
It doesn't hide the dropdown, it just disables it. There is an internal prop named `hideProjects` that does hide the namespace dropdown.